src/pybind/mgr/rbd_support/module.py

```diff
@@ -83,9 +83,13 @@ class Module(MgrModule):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         super(Module, self).__init__(*args, **kwargs)
         self.client_blocklisted = Event()
+        self.module_ready = False
+        self.mirror_snapshot_schedule = None
```

Comment:
mypy errors. Need to annotate the handler object types.

Reply:
> mypy errors. Need to annotate the handler object types.

This is just a suggestion but I would rather avoid this `None` business by adding a `setup()` method to each handler to complement an existing `shutdown()` method.  Make the constructor of the handler just initialize a few attributes, move all the actual work to `handler.setup()` and ensure that `handler.shutdown()` does the right thing regardless of whether `handler.setup()` has been called or not.

Reply:
If you decide to implement this, I would also rename `Module`'s `setup()` method to `setup_handlers()` to minimize confusion.

Reply:
The PR no longer sets handlers to None. I've tried to incorporate your comments above as best as I understood it. Let me know what you think.